src/transformers/tokenization_utils_base.py

```diff
@@ -1627,6 +1636,76 @@ def get_vocab(self) -> Dict[str, int]:
         """
         raise NotImplementedError()
 
+    def apply_chat_template(
+        self,
+        conversation: Union[List[Dict[str, str]], "Conversation"],
+        chat_template: Optional[str] = None,
+        tokenize: bool = True,
+        **tokenizer_kwargs,
```

Comment:
Not super happy about passing the whole `tokenizer_kwargs` here, can't we write out all elements that we can pass or are there too many? 

Also cc @ArthurZucker

Reply:
I can't think of a complete list here, since I'm not totally sure of which kwargs users will want! I'm not sure how to make a shorter list of specific args, so I'm tempted to leave it as `**tokenizer_kwargs` for now.

Reply:
In favor to restricting the usage by just not allowing the `tokenize = True`, WDYT?

Reply:
I think having the `tokenizer_kwargs` leads this to being more permissive to changes in the future so I think it's fine for now; but I agree with Patrick that we should try to limit `**kwargs` as much as possible so if we can be clear about what needs to be passed here it'll be clearer to the end-user as well.

You might want to isolate a few specific arguments, such as `truncation`, `padding`, etc, to make it clearer.

Here you might also want to specify that it's kwargs that will be passed to the `encode` method of the tokenizer specifically.